src/js/binaryen.js-pre.js

```diff
@@ -0,0 +1,12 @@
+// Make sure node is able to find the .wasm next to the .js file. Only relevant
+// when building binaryen-wasm.js (not used for binaryen.js currently).
+if (
```

Comment:
as this is a pre-js, you can actually just check the `ENVIRONMENT_IS_NODE` variable

Reply:
I tried, but it wasn't defined already at this point.

Reply:
Just checked again without a minifier. Appears pre-js goes right after `if (!Module) Module = typeof Binaryen !== 'undefined' ? Binaryen : {};` (starts at line 21) while `ENVIRONMENT_IS_NODE` is defined a bit later (line 63).

Reply:
Oh right, the pre-js goes before everything...

Reply:
Would it make sense to move just the `ENVIRONMENT_IS_XY` definitions up so pre-js can use/modify them? Could create a PR for this if it makes sense.

Reply:
See: https://github.com/kripken/emscripten/pull/6137